collections/src/main/scala/strawman/collection/immutable/Set.scala

```diff
@@ -9,7 +9,7 @@ import scala.{Any, Boolean, Int, deprecatedName, `inline`, None, Option, Seriali
 /** Base trait for immutable set collections */
 trait Set[A] extends Iterable[A] with collection.Set[A] with SetOps[A, Set, Set[A]] {
 
-  override final def toSet[B >: A]: Set[B] = this.asInstanceOf[Set[B]]
+  override final def toSet[B >: A]: Set[B] = Set.from(this)
```

Comment:
Why do we need to make a copy?

Reply:
Maybe we could make it a no-op in final subclasses of `Set` at least?

Reply:
`Set.from` doesn't copy when it's not necessary but we need to take "really invariant" subtypes of `Set` into account, which is what `Set.from` already does.

Reply:
Actually we can just remove this override because the base implementation in `Iterable` already does the same.